op-chain-ops/cmd/withdrawals/main.go

```diff
@@ -33,63 +87,345 @@ func main() {
 			}
 
-			backends := crossdomain.NewBackends(l1Client, l2Client)
+			// create the set of withdrawals
+			wds, err := newWithdrawals(ctx, l1ChainID)
+			if err != nil {
+				return err
+			}
 
-			l1xDomainMessenger := ctx.String("l1-cross-domain-messenger-address")
-			if l1xDomainMessenger == "" {
-				return errors.New("Must pass in L1CrossDomainMessenger address")
+			period, err := contracts.OptimismPortal.FINALIZATIONPERIODSECONDS(&bind.CallOpts{})
+			if err != nil {
+				return err
 			}
 
-			l1xDomainMessengerAddr := common.HexToAddress(l1xDomainMessenger)
-			messengers, err := crossdomain.NewMessengers(backends, l1xDomainMessengerAddr)
+			bedrockStartingBlockNumber, err := contracts.L2OutputOracle.StartingBlockNumber(&bind.CallOpts{})
 			if err != nil {
 				return err
 			}
 
-			start := ctx.Uint64("start")
-			end := ctx.Uint64("end")
+			bedrockStartingBlock, err := clients.L2Client.BlockByNumber(context.Background(), bedrockStartingBlockNumber)
+			if err != nil {
+				return err
+			}
+
+			log.Info("Withdrawal config", "finalization-period", period, "bedrock-starting-block-number", bedrockStartingBlockNumber, "bedrock-starting-block-hash", bedrockStartingBlock.Hash().Hex())
 
-			// All messages are expected to be version 0 messages
-			withdrawals, err := crossdomain.GetPendingWithdrawals(messengers, common.Big0, start, end)
+			if !bytes.Equal(bedrockStartingBlock.Extra(), genesis.BedrockTransitionBlockExtraData) {
+				return errors.New("genesis block mismatch")
+			}
+
+			outfile := ctx.String("bad-withdrawals-out")
+			f, err := os.OpenFile(outfile, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0o755)
 			if err != nil {
 				return err
 			}
 
-			outfile := ctx.String("outfile")
-			if err := writeJSONFile(outfile, withdrawals); err != nil {
+			// create a transactor
+			opts, err := newTransactor(ctx)
+			if err != nil {
 				return err
 			}
 
+			// Need this to compare in event parsing
+			l1StandardBridgeAddress := common.HexToAddress(ctx.String("l1-standard-bridge-address"))
+
+			// iterate over all of the withdrawals and submit them
+			for i, wd := range wds {
+				log.Info("Processing withdrawal", "index", i)
+
+				// migrate the withdrawal
+				withdrawal, err := crossdomain.MigrateWithdrawal(wd, &l1xdmAddr)
+				if err != nil {
+					return err
+				}
+
+				// Pass to Portal
+				hash, err := withdrawal.Hash()
+				if err != nil {
+					return err
+				}
+
+				lcdm := wd.CrossDomainMessage()
+				legacyXdmHash, err := lcdm.Hash()
+				if err != nil {
+					return err
+				}
+
+				// check to see if the withdrawal has already been successfully
+				// relayed or received
+				isSuccess, err := contracts.L1CrossDomainMessenger.SuccessfulMessages(&bind.CallOpts{}, legacyXdmHash)
+				if err != nil {
+					return err
+				}
+				isFailed, err := contracts.L1CrossDomainMessenger.FailedMessages(&bind.CallOpts{}, legacyXdmHash)
+				if err != nil {
+					return err
+				}
+
+				// This is just wrong
```

Comment:
What do you mean by "This is just wrong"? Is this a post bedrock withdrawal (the `new`)?

Reply:
ah sorry, that is a dead comment. good catch